juju/charmhub.py

```diff
@@ -18,18 +18,22 @@ def request_charmhub_with_retry(self, url, retries):
             jasyncio.sleep(5)
         raise JujuError("Got {} from {}".format(_response.status_code, url))
 
-    def get_charm_id(self, charm_name):
+    async def get_charm_id(self, charm_name):
         conn, headers, path_prefix = self.model.connection().https_connection()
 
-        url = "https://api.snapcraft.io/v2/charms/info/{}".format(charm_name)
+        model_conf = await self.model.get_config()
+        charmhub_url = model_conf['charmhub-url']
+        url = "{}/v2/charms/info/{}".format(charmhub_url.value, charm_name)
         _response = self.request_charmhub_with_retry(url, 5)
         response = json.loads(_response.text)
         return response['id'], response['name']
 
-    def is_subordinate(self, charm_name):
+    async def is_subordinate(self, charm_name):
         conn, headers, path_prefix = self.model.connection().https_connection()
 
-        url = "https://api.snapcraft.io/v2/charms/info/{}?fields=default-release.revision.subordinate".format(charm_name)
+        model_conf = await self.model.get_config()
+        charmhub_url = model_conf['charmhub-url']
+        url = "{}/v2/charms/info/{}?fields=default-release.revision.subordinate".format(charmhub_url.value, charm_name)
```

Comment:
Now that we've dropped Python 3.5 support we can use fstrings right?

Reply:
Yeah I agree :+1:  we should probably turn all the formattings into fstrings at some point